addons/mail/models/mail_thread.py

```diff
@@ -588,12 +588,14 @@ def message_change_thread(self, new_thread):
     @tools.ormcache()
     def _get_tracked_fields(self):
         """ Return the set of tracked fields names for the current model. """
-        return {
+        fields = {
```

Comment:
Mmmh, this is likely to lead to issues if you cache things depending on current user ACLs :) . To check where _get_tracked_fields is used, but you should probably give user-based conditions to the cache mechanism like
```
@tools.ormcache('self.env.uid')
```

Reply:
Maybe carrément
```
@tools.ormcache('self.env.uid', 'self.env.su')
```

Reply:
@tde-banana-odoo You are probably right (the method is used in the write/create of the mail thread, so it can be executed by different users)